Corrections

dreddsa5dies opened the pull request:
Thank you for the project! I use it in my projects.
I use golangci-lint, which reports multiple recommendations. I made a part in the form of edits. If you are interested, I can continue working. I ask you to consider PR and express your opinion.

deitch commented:
Happy to have these, thank you.

The CI already calls `make lint`, which uses golangci-lint. Can you tell why it didn't catch these? If so, let's fix that and make it all one PR with the fix.

Also, let's squash these down to one commit.

dreddsa5dies commented:
I think because it's worth checking fast in the settings. I can offer my own configuration file.

dreddsa5dies commented:
Of course! All linters except revive are disabled in the makefile!
See [this](https://github.com/diskfs/go-diskfs/blob/master/Makefile#L72)

deitch commented:
> Of course! All linters except revive are disabled in the makefile!

🤦‍♂️ 

I just hit "approve and run". Once CI passes, I will squash and merge. As soon as it is in, @dreddsa5dies , can you add a PR with lint fixing (since you already pretty much have it in hand)?

deitch commented:
> I use it in my projects.

Which projects? How do you use it? I want to know about use cases.

dreddsa5dies commented:
So. The passing of the tests has been [corrected](https://github.com/diskfs/go-diskfs/pull/148/commits/6334d78402f47e69d4e8dd398eb5a332bb35c811), it was in a different error output register in the tests and code (you need to write lowercase everywhere, [here](https://github.com/golang/go/wiki/CodeReviewComments#error-strings) are the recommendations of the Go developers). Added the config to golangci-lint and corrected it in the makefile. Sincerely yours!

deitch commented:
Yeah, some of this stuff has a long history, some of the error messages were based on other things, etc. I am most glad to have it cleaned up.

I kicked off CI again, but it appears your changes broke something else?

dreddsa5dies commented:
Of course! I added a lint check. And there are still a lot of alerts. Please see the [details](https://github.com/diskfs/go-diskfs/runs/7465292067?check_suite_focus=true#step:5:11). Your suggestions? So, you can return the lint check to its original state, but then the original [GOAL](https://github.com/diskfs/go-diskfs/pull/148#issuecomment-1185347868) will not be achieved.

deitch commented:
Can you replicate the errors? I cloned your branch and ran `make lint`, and get a completely different set. 🤷‍♂️

deitch commented:
The errors also are odd to me. See, e.g. these 4 from `diskfs_test.go`:

```
Error: diskfs_test.go:69:13: undeclared name: `diskfs` (typecheck)
[11](https://github.com/diskfs/go-diskfs/runs/7465292067?check_suite_focus=true#step:5:12)
		d, err := diskfs.Open(tt.path)
[12](https://github.com/diskfs/go-diskfs/runs/7465292067?check_suite_focus=true#step:5:13)
		          ^
[13](https://github.com/diskfs/go-diskfs/runs/7465292067?check_suite_focus=true#step:5:14)
Error: diskfs_test.go:84:13: undeclared name: `diskfs` (typecheck)
[14](https://github.com/diskfs/go-diskfs/runs/7465292067?check_suite_focus=true#step:5:15)
		d, err := diskfs.Open(tt.path, diskfs.WithOpenMode(diskfs.ReadOnly))
[15](https://github.com/diskfs/go-diskfs/runs/7465292067?check_suite_focus=true#step:5:16)
		          ^
[16](https://github.com/diskfs/go-diskfs/runs/7465292067?check_suite_focus=true#step:5:17)
Error: diskfs_test.go:103:14: undeclared name: `diskfs` (typecheck)
[17](https://github.com/diskfs/go-diskfs/runs/7465292067?check_suite_focus=true#step:5:18)
		format     diskfs.Format
[18](https://github.com/diskfs/go-diskfs/runs/7465292067?check_suite_focus=true#step:5:19)
		           ^
[19](https://github.com/diskfs/go-diskfs/runs/7465292067?check_suite_focus=true#step:5:20)
Error: diskfs_test.go:10:2: "github.com/diskfs/go-diskfs" imported but not used (typecheck)
[20](https://github.com/diskfs/go-diskfs/runs/7465292067?check_suite_focus=true#step:5:21)
	"github.com/diskfs/go-diskfs"
[21](https://github.com/diskfs/go-diskfs/runs/7465292067?check_suite_focus=true#step:5:22)
	^
[22](https://github.com/diskfs/go-diskfs/runs/7465292067?check_suite_focus=true#step:5:23)
```

It is complaining that it didn't use `go-diskfs` and then complaining that `diskfs` is undeclared. Similar errors abound.

dreddsa5dies commented:
```make lint``` outputs only part of the alerts, then ends. 
I don't find what you are reporting ("undeclared").
<details><summary>List of alerts</summary>

```go
diskfs_test.go:71: 71-80 lines are duplicate of `diskfs_test.go:86-95` (dupl)
		switch {
		case (err == nil && tt.err != nil) || (err != nil && tt.err == nil) || (err != nil && tt.err != nil && !strings.HasPrefix(err.Error(), tt.err.Error())):
			t.Errorf("%s: mismatched errors, actual %v expected %v", msg, err, tt.err)
		case (d == nil && tt.disk != nil) || (d != nil && tt.disk == nil):
			t.Errorf("%s: mismatched disk, actual %v expected %v", msg, d, tt.disk)
		case d != nil && (d.LogicalBlocksize != tt.disk.LogicalBlocksize || d.PhysicalBlocksize != tt.disk.PhysicalBlocksize || d.Size != tt.disk.Size || d.Type != tt.disk.Type):
			t.Errorf("%s: mismatched disk, actual then expected", msg)
			t.Logf("%v", d)
			t.Logf("%v", tt.disk)
		}
diskfs_test.go:86: 86-95 lines are duplicate of `diskfs_test.go:120-129` (dupl)
		switch {
		case (err == nil && tt.err != nil) || (err != nil && tt.err == nil) || (err != nil && tt.err != nil && !strings.HasPrefix(err.Error(), tt.err.Error())):
			t.Errorf("%s: mismatched errors, actual %v expected %v", msg, err, tt.err)
		case (d == nil && tt.disk != nil) || (d != nil && tt.disk == nil):
			t.Errorf("%s: mismatched disk, actual %v expected %v", msg, d, tt.disk)
		case d != nil && (d.LogicalBlocksize != tt.disk.LogicalBlocksize || d.PhysicalBlocksize != tt.disk.PhysicalBlocksize || d.Size != tt.disk.Size || d.Type != tt.disk.Type):
			t.Errorf("%s: mismatched disk, actual then expected", msg)
			t.Logf("%v", d)
			t.Logf("%v", tt.disk)
		}
diskfs_test.go:120: 120-129 lines are duplicate of `diskfs_test.go:71-80` (dupl)
		switch {
		case (err == nil && tt.err != nil) || (err != nil && tt.err == nil) || (err != nil && tt.err != nil && !strings.HasPrefix(err.Error(), tt.err.Error())):
			t.Errorf("%s: mismatched errors, actual %v expected %v", msg, err, tt.err)
		case (disk == nil && tt.disk != nil) || (disk != nil && tt.disk == nil):
			t.Errorf("%s: mismatched disk, actual %v expected %v", msg, disk, tt.disk)
		case disk != nil && (disk.LogicalBlocksize != tt.disk.LogicalBlocksize || disk.PhysicalBlocksize != tt.disk.PhysicalBlocksize || disk.Size != tt.disk.Size || disk.Type != tt.disk.Type):
			t.Errorf("%s: mismatched disk, actual then expected", msg)
			t.Logf("%#v", disk)
			t.Logf("%#v", tt.disk)
		}
diskfs_test.go:26:13: Error return value of `f.Truncate` is not checked (errcheck)
		f.Truncate(10 * 1024 * 1024)
		          ^
diskfs_test.go:118:3: importShadow: shadow of imported from 'github.com/diskfs/go-diskfs/disk' package 'disk' (gocritic)
		disk, err := diskfs.Create(tt.path, tt.size, tt.format, tt.sectorSize)
		^
diskfs_test.go:44:1: cyclomatic complexity 37 of func `TestOpen` is high (> 30) (gocyclo)
func TestOpen(t *testing.T) {
^
disk/disk_test.go:337:5: commentedOutCode: may want to remove commented-out code (gocritic)
				// t.Log(b)
				^
disk/disk_test.go:217:4: deferInLoop: Possible resource leak, 'defer' is called in the 'for' loop (gocritic)
			defer f.Close()
			^
disk/disk_test.go:220:5: deferInLoop: Possible resource leak, 'defer' is called in the 'for' loop (gocritic)
				defer os.Remove(f.Name())
				^
disk/disk_test.go:296:4: deferInLoop: Possible resource leak, 'defer' is called in the 'for' loop (gocritic)
			defer f.Close()
			^
disk/disk_test.go:267:1: cyclomatic complexity 33 of func `TestReadPartitionContents` is high (> 30) (gocyclo)
func TestReadPartitionContents(t *testing.T) {
^
disk/disk_test.go:89: unnecessary trailing newline (whitespace)

}
disk/disk_test.go:408:26: S1004: should use !bytes.Equal(b, b2) instead (gosimple)
			case tt.err == nil && bytes.Compare(b, b2) != 0:
			                      ^
filesystem/iso9660/eltorito.go:14:2: `elToritoSector` is unused (deadcode)
	elToritoSector        = 0x11
	^
filesystem/iso9660/eltorito.go:15:2: `elToritoDefaultBlocks` is unused (deadcode)
	elToritoDefaultBlocks = 4
	^
filesystem/iso9660/finalize.go:23:6: `fileInfoFinder` is unused (deadcode)
type fileInfoFinder interface {
     ^
filesystem/iso9660/volume_descriptor_internal_test.go:80:6: `getValidVolumeDescriptors` is unused (deadcode)
func getValidVolumeDescriptors() ([]volumeDescriptor, []byte, error) {
     ^
filesystem/iso9660/iso9660_internal_test.go:30:2: declaration has 3 blank identifiers (dogsled)
	validDe, _, _, _, err := get9660DirectoryEntries(fs)
	^
filesystem/iso9660/iso9660_internal_test.go:92:2: declaration has 3 blank identifiers (dogsled)
	validDe, _, _, _, err := getRockRidgeDirectoryEntries(fs, false)
	^
filesystem/iso9660/pathtable_internal_test.go:130:2: declaration has 3 blank identifiers (dogsled)
	table, _, _, _ := get9660PathTable()
	^
filesystem/iso9660/pathtable_internal_test.go:10: 10-54 lines are duplicate of `filesystem/iso9660/pathtable_internal_test.go:56-100` (dupl)
func get9660PathTable() (*pathTable, []byte, []byte, error) {
	blocksize := 2048
	// sector 27 - L path table
	// sector 28 - M path table
	pathTableLSector := 35
	pathTableMSector := 36
	// read correct bytes off of disk
	input, err := os.ReadFile(ISO9660File)
	if err != nil {
		return nil, nil, nil, fmt.Errorf("Error reading data from iso9660 test fixture %s: %v", ISO9660File, err)
	}

	startL := pathTableLSector * blocksize // start of pathtable in file.iso

	// one block, since we know it is just one block
	LBytes := input[startL : startL+blocksize]

	startM := pathTableMSector * blocksize // start of pathtable in file.iso

	// one block, since we know it is just one block
	MBytes := input[startM : startM+blocksize]

	entries := []*pathTableEntry{
		{nameSize: 0x1, size: 0xa, extAttrLength: 0x0, location: 0x12, parentIndex: 0x1, dirname: "\x00"},
		{nameSize: 0x3, size: 0xc, extAttrLength: 0x0, location: 0x13, parentIndex: 0x1, dirname: "ABC"},
		{nameSize: 0x3, size: 0xc, extAttrLength: 0x0, location: 0x14, parentIndex: 0x1, dirname: "BAR"},
		{nameSize: 0x4, size: 0xc, extAttrLength: 0x0, location: 0x15, parentIndex: 0x1, dirname: "DEEP"},
		{nameSize: 0x3, size: 0xc, extAttrLength: 0x0, location: 0x21, parentIndex: 0x1, dirname: "FOO"},
		{nameSize: 0x1, size: 0xa, extAttrLength: 0x0, location: 0x16, parentIndex: 0x4, dirname: "A"},
		{nameSize: 0x1, size: 0xa, extAttrLength: 0x0, location: 0x17, parentIndex: 0x6, dirname: "B"},
		{nameSize: 0x1, size: 0xa, extAttrLength: 0x0, location: 0x18, parentIndex: 0x7, dirname: "C"},
		{nameSize: 0x1, size: 0xa, extAttrLength: 0x0, location: 0x19, parentIndex: 0x8, dirname: "D"},
		{nameSize: 0x1, size: 0xa, extAttrLength: 0x0, location: 0x1a, parentIndex: 0x9, dirname: "E"},
		{nameSize: 0x1, size: 0xa, extAttrLength: 0x0, location: 0x1b, parentIndex: 0xa, dirname: "F"},
		{nameSize: 0x1, size: 0xa, extAttrLength: 0x0, location: 0x1c, parentIndex: 0xb, dirname: "G"},
		{nameSize: 0x1, size: 0xa, extAttrLength: 0x0, location: 0x1d, parentIndex: 0xc, dirname: "H"},
		{nameSize: 0x1, size: 0xa, extAttrLength: 0x0, location: 0x1e, parentIndex: 0xd, dirname: "I"},
		{nameSize: 0x1, size: 0xa, extAttrLength: 0x0, location: 0x1f, parentIndex: 0xe, dirname: "J"},
		{nameSize: 0x1, size: 0xa, extAttrLength: 0x0, location: 0x20, parentIndex: 0xf, dirname: "K"},
	}

	return &pathTable{
		records: entries,
	}, LBytes, MBytes, nil
}
filesystem/iso9660/pathtable_internal_test.go:56: 56-100 lines are duplicate of `filesystem/iso9660/pathtable_internal_test.go:10-54` (dupl)
func getRockRidgePathTable() (*pathTable, []byte, []byte, error) {
	blocksize := 2048
	// sector 27 - L path table
	// sector 28 - M path table
	pathTableLSector := 39
	pathTableMSector := 40
	// read correct bytes off of disk
	input, err := os.ReadFile(RockRidgeFile)
	if err != nil {
		return nil, nil, nil, fmt.Errorf("Error reading data from iso9660 test fixture %s: %v", RockRidgeFile, err)
	}

	startL := pathTableLSector * blocksize // start of pathtable in file.iso

	// one block, since we know it is just one block
	LBytes := input[startL : startL+blocksize]

	startM := pathTableMSector * blocksize // start of pathtable in file.iso

	// one block, since we know it is just one block
	MBytes := input[startM : startM+blocksize]

	entries := []*pathTableEntry{
		{nameSize: 0x1, size: 0xa, extAttrLength: 0x0, location: 0x12, parentIndex: 0x1, dirname: "\x00"},
		{nameSize: 0x3, size: 0xc, extAttrLength: 0x0, location: 0x14, parentIndex: 0x1, dirname: "ABC"},
		{nameSize: 0x3, size: 0xc, extAttrLength: 0x0, location: 0x15, parentIndex: 0x1, dirname: "BAR"},
		{nameSize: 0x4, size: 0xc, extAttrLength: 0x0, location: 0x16, parentIndex: 0x1, dirname: "DEEP"},
		{nameSize: 0x3, size: 0xc, extAttrLength: 0x0, location: 0x1d, parentIndex: 0x1, dirname: "FOO"},
		{nameSize: 0x1, size: 0xa, extAttrLength: 0x0, location: 0x22, parentIndex: 0x1, dirname: "G"},
		{nameSize: 0x1, size: 0xa, extAttrLength: 0x0, location: 0x17, parentIndex: 0x4, dirname: "A"},
		{nameSize: 0x1, size: 0xa, extAttrLength: 0x0, location: 0x23, parentIndex: 0x6, dirname: "H"},
		{nameSize: 0x1, size: 0xa, extAttrLength: 0x0, location: 0x18, parentIndex: 0x7, dirname: "B"},
		{nameSize: 0x1, size: 0xa, extAttrLength: 0x0, location: 0x24, parentIndex: 0x8, dirname: "I"},
		{nameSize: 0x1, size: 0xa, extAttrLength: 0x0, location: 0x19, parentIndex: 0x9, dirname: "C"},
		{nameSize: 0x1, size: 0xa, extAttrLength: 0x0, location: 0x25, parentIndex: 0xa, dirname: "J"},
		{nameSize: 0x1, size: 0xa, extAttrLength: 0x0, location: 0x1a, parentIndex: 0xb, dirname: "D"},
		{nameSize: 0x1, size: 0xa, extAttrLength: 0x0, location: 0x26, parentIndex: 0xc, dirname: "K"},
		{nameSize: 0x1, size: 0xa, extAttrLength: 0x0, location: 0x1b, parentIndex: 0xd, dirname: "E"},
		{nameSize: 0x1, size: 0xa, extAttrLength: 0x0, location: 0x1c, parentIndex: 0xf, dirname: "F"},
	}

	return &pathTable{
		records: entries,
	}, LBytes, MBytes, nil
}
filesystem/iso9660/file.go:50:13: Error return value of `file.ReadAt` is not checked (errcheck)
	file.ReadAt(b[0:maxRead], int64(location)*fs.blocksize+int64(fl.offset))
	           ^
filesystem/iso9660/finalize.go:595:13: Error return value of `f.WriteAt` is not checked (errcheck)
			f.WriteAt(e, writeAt+int64(i*blocksize))
			         ^
filesystem/iso9660/finalize.go:601:11: Error return value of `f.WriteAt` is not checked (errcheck)
	f.WriteAt(pathTableLBytes, writeAt)
	         ^
filesystem/iso9660/finalize.go:603:11: Error return value of `f.WriteAt` is not checked (errcheck)
	f.WriteAt(pathTableMBytes, writeAt)
	         ^
filesystem/iso9660/finalize.go:822:10: Error return value of `os.Chdir` is not checked (errcheck)
	os.Chdir(workspace)
	        ^
filesystem/iso9660/finalize.go:865:10: Error return value of `os.Chdir` is not checked (errcheck)
	os.Chdir(cwd)
	        ^
filesystem/iso9660/iso9660.go:187:10: Error return value is not checked (errcheck)
			pvd = vd.(*primaryVolumeDescriptor)
			      ^
filesystem/iso9660/directoryentry.go:87:3: assignOp: replace `flagByte = flagByte | 0x01` with `flagByte |= 0x01` (gocritic)
		flagByte = flagByte | 0x01
		^
filesystem/iso9660/directoryentry.go:90:3: assignOp: replace `flagByte = flagByte | 0x02` with `flagByte |= 0x02` (gocritic)
		flagByte = flagByte | 0x02
		^
filesystem/iso9660/directoryentry.go:93:3: assignOp: replace `flagByte = flagByte | 0x04` with `flagByte |= 0x04` (gocritic)
		flagByte = flagByte | 0x04
		^
filesystem/iso9660/directoryentry.go:96:3: assignOp: replace `flagByte = flagByte | 0x08` with `flagByte |= 0x08` (gocritic)
		flagByte = flagByte | 0x08
		^
filesystem/iso9660/directoryentry.go:99:3: assignOp: replace `flagByte = flagByte | 0x10` with `flagByte |= 0x10` (gocritic)
		flagByte = flagByte | 0x10
		^
filesystem/iso9660/directoryentry.go:102:3: assignOp: replace `flagByte = flagByte | 0x80` with `flagByte |= 0x80` (gocritic)
		flagByte = flagByte | 0x80
		^
filesystem/iso9660/directoryentry.go:540:2: commentedOutCode: may want to remove commented-out code (gocritic)
	// 		return nil, fmt.Errorf("Invalid d-character")
	^
filesystem/iso9660/directoryentry.go:486:9: octalLiteral: use new octal literal style, 0o755 (gocritic)
	return 0755
	       ^
filesystem/iso9660/directoryentry.go:366:1: unnamedResult: consider giving a name to these results (gocritic)
func (de *directoryEntry) getLocation(p string) (uint32, uint32, error) {
^
filesystem/iso9660/directoryentry.go:553:12: wrapperFunc: use strings.ReplaceAll method in `strings.Replace(s, ".", "", -1)` (gocritic)
		case len(strings.Replace(s, ".", "", -1)) > 30:
		         ^
filesystem/iso9660/directoryentrysystemuseextension.go:83:2: stringXbytes: can simplify `[]byte(d.Signature())` to `d.Signature()` (gocritic)
	copy(ret[0:2], []byte(d.Signature()))
	^
filesystem/iso9660/directoryentrysystemuseextension.go:309:2: stringXbytes: can simplify `[]byte(suspExtensionContinuationArea)` to `suspExtensionContinuationArea` (gocritic)
	copy(ret[0:2], []byte(suspExtensionContinuationArea))
	^
filesystem/iso9660/directoryentrysystemuseextension.go:446:2: stringXbytes: can simplify `[]byte(suspExtensionExtensionsReference)` to `suspExtensionExtensionsReference` (gocritic)
	copy(ret[0:2], []byte(suspExtensionExtensionsReference))
	^
filesystem/iso9660/directoryentrysystemuseextension.go:377:2: stringXbytes: can simplify `[]byte(suspExtensionExtensionsSelector)` to `suspExtensionExtensionsSelector` (gocritic)
	copy(ret[0:2], []byte(suspExtensionExtensionsSelector))
	^
filesystem/iso9660/directoryentrysystemuseextension.go:200:2: stringXbytes: can simplify `[]byte(suspExtensionPaddingField)` to `suspExtensionPaddingField` (gocritic)
	copy(ret[0:2], []byte(suspExtensionPaddingField))
	^
filesystem/iso9660/directoryentrysystemuseextension.go:138:2: stringXbytes: can simplify `[]byte(suspExtensionSharingProtocolIndicator)` to `suspExtensionSharingProtocolIndicator` (gocritic)
	copy(ret[0:2], []byte(suspExtensionSharingProtocolIndicator))
	^
filesystem/iso9660/directoryentrysystemuseextension.go:249:2: stringXbytes: can simplify `[]byte(suspExtensionSharingProtocolTerminator)` to `suspExtensionSharingProtocolTerminator` (gocritic)
	copy(ret[0:2], []byte(suspExtensionSharingProtocolTerminator))
	^
filesystem/iso9660/eltorito.go:100:2: stringXbytes: can simplify `[]byte(util.AppNameVersion)` to `util.AppNameVersion` (gocritic)
	copy(b[4:0x1c], []byte(util.AppNameVersion))
	^
filesystem/iso9660/file.go:52:2: assignOp: replace `fl.offset = fl.offset + int64(maxRead)` with `fl.offset += int64(maxRead)` (gocritic)
	fl.offset = fl.offset + int64(maxRead)
	^
filesystem/iso9660/finalize.go:193:1: unnamedResult: consider giving a name to these results (gocritic)
func (fi *finalizeFileInfo) calculateRecordSize(fs *FileSystem, isSelf, isParent bool) (int, int, error) {
^
filesystem/iso9660/finalize.go:210:1: unnamedResult: consider giving a name to these results (gocritic)
func (fi *finalizeFileInfo) calculateDirectorySize(fs *FileSystem) (int, int, error) {
^
filesystem/iso9660/iso9660.go:324:13: ioutilDeprecated: ioutil.ReadDir is deprecated, use os.ReadDir instead (gocritic)
		fi, err = ioutil.ReadDir(fullPath)
		          ^
filesystem/iso9660/iso9660.go:303:49: octalLiteral: use new octal literal style, 0o755 (gocritic)
	err := os.MkdirAll(path.Join(fs.workspace, p), 0755)
	                                               ^
filesystem/iso9660/iso9660.go:405:58: octalLiteral: use new octal literal style, 0o644 (gocritic)
		f, err = os.OpenFile(path.Join(fs.workspace, p), flag, 0644)
		                                                       ^
filesystem/iso9660/iso9660.go:64:1: paramTypeCombine: func(f util.File, size int64, start int64, blocksize int64, workspace string) (*FileSystem, error) could be replaced with func(f util.File, size, start, blocksize int64, workspace string) (*FileSystem, error) (gocritic)
func Create(f util.File, size int64, start int64, blocksize int64, workspace string) (*FileSystem, error) {
^
filesystem/iso9660/iso9660.go:128:1: paramTypeCombine: func(file util.File, size int64, start int64, blocksize int64) (*FileSystem, error) could be replaced with func(file util.File, size, start, blocksize int64) (*FileSystem, error) (gocritic)
func Read(file util.File, size int64, start int64, blocksize int64) (*FileSystem, error) {
^
filesystem/iso9660/rockridge.go:850:3: assignOp: replace `b[4] = b[4] | rockRidgeTimestampLongForm` with `b[4] |= rockRidgeTimestampLongForm` (gocritic)
		b[4] = b[4] | rockRidgeTimestampLongForm
		^
filesystem/iso9660/rockridge.go:860:3: assignOp: replace `b[4] = b[4] | t.timestampType` with `b[4] |= t.timestampType` (gocritic)
		b[4] = b[4] | t.timestampType
		^
filesystem/iso9660/rockridge.go:605:4: assignOp: replace `d.name = d.name + l.name` with `d.name += l.name` (gocritic)
			d.name = d.name + l.name
			^
filesystem/iso9660/rockridge.go:722:4: assignOp: replace `d.name = d.name + n.name` with `d.name += n.name` (gocritic)
			d.name = d.name + n.name
			^
filesystem/iso9660/rockridge.go:705:4: assignOp: replace `flags = flags | 0x2` with `flags |= 0x2` (gocritic)
			flags = flags | 0x2
			^
filesystem/iso9660/rockridge.go:708:4: assignOp: replace `flags = flags | 0x4` with `flags |= 0x4` (gocritic)
			flags = flags | 0x4
			^
filesystem/iso9660/rockridge.go:405:2: assignOp: replace `m = m | (modes & 0777)` with `m |= (modes & 0777)` (gocritic)
	m = m | (modes & 0777)
	^
filesystem/iso9660/rockridge.go:408:2: assignOp: replace `m = m | (modes & uint32(os.ModeSetgid))` with `m |= (modes & uint32(os.ModeSetgid))` (gocritic)
	m = m | (modes & uint32(os.ModeSetgid))
	^
filesystem/iso9660/rockridge.go:407:2: assignOp: replace `m = m | (modes & uint32(os.ModeSetuid))` with `m |= (modes & uint32(os.ModeSetuid))` (gocritic)
	m = m | (modes & uint32(os.ModeSetuid))
	^
filesystem/iso9660/rockridge.go:421:3: assignOp: replace `m = m | uint32(os.ModeCharDevice|os.ModeDevice)` with `m |= uint32(os.ModeCharDevice|os.ModeDevice)` (gocritic)
		m = m | uint32(os.ModeCharDevice|os.ModeDevice)
		^
filesystem/iso9660/rockridge.go:423:3: assignOp: replace `m = m | uint32(os.ModeDevice)` with `m |= uint32(os.ModeDevice)` (gocritic)
		m = m | uint32(os.ModeDevice)
		^
filesystem/iso9660/rockridge.go:425:3: assignOp: replace `m = m | uint32(os.ModeDir)` with `m |= uint32(os.ModeDir)` (gocritic)
		m = m | uint32(os.ModeDir)
		^
filesystem/iso9660/rockridge.go:427:3: assignOp: replace `m = m | uint32(os.ModeNamedPipe)` with `m |= uint32(os.ModeNamedPipe)` (gocritic)
		m = m | uint32(os.ModeNamedPipe)
		^
filesystem/iso9660/rockridge.go:417:3: assignOp: replace `m = m | uint32(os.ModeSocket)` with `m |= uint32(os.ModeSocket)` (gocritic)
		m = m | uint32(os.ModeSocket)
		^
filesystem/iso9660/rockridge.go:419:3: assignOp: replace `m = m | uint32(os.ModeSymlink)` with `m |= uint32(os.ModeSymlink)` (gocritic)
		m = m | uint32(os.ModeSymlink)
		^
filesystem/iso9660/rockridge.go:356:3: assignOp: replace `modes = modes | 0100000` with `modes |= 0100000` (gocritic)
		modes = modes | 0100000
		^
filesystem/iso9660/rockridge.go:352:3: assignOp: replace `modes = modes | 010000` with `modes |= 010000` (gocritic)
		modes = modes | 010000
		^
filesystem/iso9660/rockridge.go:328:3: assignOp: replace `modes = modes | 01000` with `modes |= 01000` (gocritic)
		modes = modes | 01000
		^
filesystem/iso9660/rockridge.go:336:3: assignOp: replace `modes = modes | 0120000` with `modes |= 0120000` (gocritic)
		modes = modes | 0120000
		^
filesystem/iso9660/finalize.go:347:1: cyclomatic complexity 54 of func `(*FileSystem).Finalize` is high (> 30) (gocyclo)
func (fs *FileSystem) Finalize(options FinalizeOptions) error {
^
filesystem/iso9660/statt_others.go:1: File is not `gofmt`-ed with `-s` (gofmt)
// +build !windows
filesystem/iso9660/eltorito.go:146:41: `critiera` is a misspelling of `criteria` (misspell)
	// b[0xd:] is vendor unique selectiomn critiera. We do not yet support it, so leave as 0.
	                                       ^
filesystem/iso9660/volume_descriptor.go:66:2: `volumeFlags` is unused (structcheck)
	volumeFlags                uint8
	^
filesystem/iso9660/finalize.go:63:2: `bytes` is unused (structcheck)
	bytes              [][]byte
	^
filesystem/iso9660/finalize.go:49:2: `target` is unused (structcheck)
	target             string
	^
filesystem/iso9660/finalize.go:55:2: `recordSize` is unused (structcheck)
	recordSize         uint8
	^
filesystem/iso9660/statt_others.go:13:37: unnecessary conversion (unconvert)
			return uint32(stat.Nlink), uint32(stat.Uid), uint32(stat.Gid)
			                                 ^
filesystem/iso9660/pathtable.go:132:20: unnecessary conversion (unconvert)
		nameSize := uint8(b[i])
		                 ^
filesystem/iso9660/pathtable.go:141:23: unnecessary conversion (unconvert)
		extAttrSize := uint8(b[i+1])
		                    ^
filesystem/iso9660/directoryentry_internal_test.go:27:52: `compareDirectoryEntries` - `compareDates` always receives `false` (unparam)
func compareDirectoryEntries(a, b *directoryEntry, compareDates, compareExtensions bool) bool {
                                                   ^
filesystem/iso9660/directoryentrysystemuseextension.go:96:78: parseSystemUseExtensionRaw - result 1 (error) is always nil (unparam)
func parseSystemUseExtensionRaw(b []byte) (directoryEntrySystemUseExtension, error) {
                                                                             ^
filesystem/iso9660/eltorito.go:83:48: (*ElTorito).generateCatalog - result 1 (error) is always nil (unparam)
func (et *ElTorito) generateCatalog() ([]byte, error) {
                                               ^
filesystem/iso9660/pathtable.go:128:44: parsePathTable - result 1 (error) is always nil (unparam)
func parsePathTable(b []byte) (*pathTable, error) {
                                           ^
filesystem/iso9660/util.go:18:42: universalizePath - result 1 (error) is always nil (unparam)
func universalizePath(p string) (string, error) {
                                         ^
filesystem/iso9660/directoryentry.go:327: unnecessary trailing newline (whitespace)

}
filesystem/iso9660/iso9660.go:182:3: missing cases in switch of type volumeDescriptorType: volumeDescriptorBoot, volumeDescriptorPartition, volumeDescriptorSupplementary (exhaustive)
		switch vd.Type() {
		^
filesystem/iso9660/directoryentry.go:407:7: S1023: redundant break statement (gosimple)
						break
						^
filesystem/iso9660/directoryentry.go:465:5: S1023: redundant break statement (gosimple)
				break
				^
filesystem/iso9660/finalize_internal_test.go:50:22: SA1019: os.SEEK_SET has been deprecated since Go 1.7: Use io.SeekStart, io.SeekCurrent, and io.SeekEnd. (staticcheck)
	_, err = to.Seek(0, os.SEEK_SET)
	                    ^
filesystem/iso9660/iso9660_internal_test.go:13:2: SA5001: should check returned error before deferring file.Close() (staticcheck)
	defer file.Close()
	^
filesystem/iso9660/iso9660_internal_test.go:73:2: SA5001: should check returned error before deferring file.Close() (staticcheck)
	defer file.Close()
	^
filesystem/iso9660/directoryentry.go:127:16: ST1005: error strings should not be capitalized (stylecheck)
			return nil, fmt.Errorf("Invalid %s %s: %v", nametype, de.filename, err)
			            ^
filesystem/iso9660/directoryentry.go:131:16: ST1005: error strings should not be capitalized (stylecheck)
			return nil, fmt.Errorf("Error converting filename to bytes: %v", err)
			            ^
filesystem/iso9660/directoryentry.go:143:16: ST1005: error strings should not be capitalized (stylecheck)
			return nil, fmt.Errorf("Unable to convert directory entry SUSP extensions to bytes: %v", err)
			            ^
partition/gpt/table_test.go:77:6: `compareGPTBytes` is unused (deadcode)
func compareGPTBytes(b1, b2 []byte) bool {
     ^
partition/gpt/table_test.go:41:13: Error return value of `f.Truncate` is not checked (errcheck)
		f.Truncate(size)
		          ^
partition/gpt/table_test.go:380:11: Error return value of `rand.Read` is not checked (errcheck)
	rand.Read(b2)
	         ^
partition/gpt/table_test.go:406:11: Error return value of `rand.Read` is not checked (errcheck)
	rand.Read(b)
	         ^
partition/gpt/table_test.go:159:1: cyclomatic complexity 31 of func `TestTableWrite` is high (> 30) (gocyclo)
func TestTableWrite(t *testing.T) {
^
partition/gpt/table_test.go:201:30: S1019: should use make([]byte, gptSize + 512*2) instead (gosimple)
		firstBytes := make([]byte, gptSize+512*2, gptSize+512*2)
		                           ^
partition/gpt/table_test.go:379:21: S1019: should use make([]byte, size) instead (gosimple)
	b2 := make([]byte, size, size)
	                   ^
partition/gpt/table_test.go:405:20: S1019: should use make([]byte, size) instead (gosimple)
	b := make([]byte, size, size)
	                  ^
partition/gpt/table_test.go:72:3: S1004: should use bytes.Equal(b1[8:12], b2[8:12]) instead (gosimple)
		bytes.Compare(b1[8:12], b2[8:12]) == 0 &&
		^
partition/gpt/table_test.go:73:3: S1004: should use bytes.Equal(b1[12:16], b2[12:16]) instead (gosimple)
		bytes.Compare(b1[12:16], b2[12:16]) == 0 &&
		^
partition/gpt/table_test.go:74:3: S1004: should use bytes.Equal(b1[16:], b2[16:]) instead (gosimple)
		bytes.Compare(b1[16:], b2[16:]) == 0
		^
partition/gpt/table_test.go:83:21: S1004: should use bytes.Equal(b1[512:], b2[512:]) instead (gosimple)
	gptSectionMatch := bytes.Compare(b1[512:], b2[512:]) == 0
	                   ^
partition/gpt/table_test.go:84:23: S1004: should use bytes.Equal(mbr1[446+16:], mbr2[446+16:]) instead (gosimple)
	mbrPostPart1Match := bytes.Compare(mbr1[446+16:], mbr2[446+16:]) == 0
	                     ^
partition/gpt/table_test.go:88:19: S1004: should use bytes.Equal(part1[8:12], part2[8:12]) instead (gosimple)
	startLBAMatch := bytes.Compare(part1[8:12], part2[8:12]) == 0
	                 ^
partition/gpt/table_test.go:89:17: S1004: should use bytes.Equal(part1[12:16], part2[12:16]) instead (gosimple)
	endLBAMatch := bytes.Compare(part1[12:16], part2[12:16]) == 0
	               ^
partition/gpt/table_test.go:236:6: S1004: should use !bytes.Equal(tablePrimaryHeader, primaryHeader) instead (gosimple)
		if bytes.Compare(tablePrimaryHeader, primaryHeader) != 0 {
		   ^
partition/gpt/table_test.go:239:6: S1004: should use !bytes.Equal(tablePrimaryArray, primaryArray) instead (gosimple)
		if bytes.Compare(tablePrimaryArray, primaryArray) != 0 {
		   ^
partition/gpt/table_test.go:395:5: S1004: should use !bytes.Equal(b.Bytes(), b2) instead (gosimple)
	if bytes.Compare(b.Bytes(), b2) != 0 {
	   ^
partition/gpt/table_test.go:422:5: S1004: should use !bytes.Equal(b2, b) instead (gosimple)
	if bytes.Compare(b2, b) != 0 {
	   ^
partition/gpt/table_test.go:197:3: SA5001: should check returned error before deferring gpt.Close() (staticcheck)
		defer gpt.Close()
		^
partition/gpt/partition_internal_test.go:24:12: Error return value of `rand.Read` is not checked (errcheck)
		rand.Read(b)
		         ^
partition/gpt/partiton.go:215:13: Error return value of `out.Write` is not checked (errcheck)
			out.Write(b[:read])
			         ^
partition/gpt/table_internal_test.go:50:1: cyclomatic complexity 33 of func `TestTableFromBytes` is high (> 30) (gocyclo)
func TestTableFromBytes(t *testing.T) {
^
partition/gpt/partition_internal_test.go:76: unnecessary trailing newline (whitespace)

	})
partition/gpt/partition_internal_test.go:345:21: S1019: should use make([]byte, size) instead (gosimple)
		b := make([]byte, size, size)
		                  ^
partition/gpt/partition_internal_test.go:97:15: S1039: unnecessary use of fmt.Sprintf (gosimple)
		expected := fmt.Sprintf("Unable to parse partition identifier GUID")
		            ^
partition/gpt/partition_internal_test.go:118:15: S1039: unnecessary use of fmt.Sprintf (gosimple)
		expected := fmt.Sprintf("Unable to parse partition type GUID")
		            ^
partition/gpt/table_internal_test.go:79:15: S1039: unnecessary use of fmt.Sprintf (gosimple)
		expected := fmt.Sprintf("Invalid EFI Signature")
		            ^
partition/gpt/table.go:164:5: S1004: should use !bytes.Equal(b[size-2:], getMbrSignature()) instead (gosimple)
	if bytes.Compare(b[size-2:], getMbrSignature()) != 0 {
	   ^
partition/gpt/table.go:357:5: S1004: should use !bytes.Equal(efiSignature, getEfiSignature()) instead (gosimple)
	if bytes.Compare(efiSignature, getEfiSignature()) != 0 {
	   ^
partition/gpt/table.go:360:5: S1004: should use !bytes.Equal(efiRevision, getEfiRevision()) instead (gosimple)
	if bytes.Compare(efiRevision, getEfiRevision()) != 0 {
	   ^
partition/gpt/table.go:363:5: S1004: should use !bytes.Equal(efiHeaderSize, getEfiHeaderSize()) instead (gosimple)
	if bytes.Compare(efiHeaderSize, getEfiHeaderSize()) != 0 {
	   ^
partition/gpt/table.go:366:5: S1004: should use !bytes.Equal(efiZeroes, getEfiZeroes()) instead (gosimple)
	if bytes.Compare(efiZeroes, getEfiZeroes()) != 0 {
	   ^
partition/gpt/partition_internal_test.go:164:6: S1004: should use !bytes.Equal(expected, b) instead (gosimple)
		if bytes.Compare(expected, b) != 0 {
		   ^
partition/gpt/partition_internal_test.go:428:6: S1004: should use !bytes.Equal(b2, b) instead (gosimple)
		if bytes.Compare(b2, b) != 0 {
		   ^
partition/gpt/partition_internal_test.go:493:6: S1004: should use !bytes.Equal(b.Bytes(), b2) instead (gosimple)
		if bytes.Compare(b.Bytes(), b2) != 0 {
		   ^
partition/gpt/partiton.go:255:7: SA4003: every value of type uint64 is >= 0 (staticcheck)
	case start >= 0 && end >= start && size == calculatedSize:
	     ^
partition/gpt/partiton.go:256:20: SA4003: every value of type uint64 is >= 0 (staticcheck)
	case size == 0 && start >= 0 && end >= start:
	                  ^
partition/gpt/table.go:235:15: SA4010: this result of append is never used, except maybe in other appends (staticcheck)
		realParts = append(realParts, part)
		            ^
partition/gpt/partition_internal_test.go:73:6: SA5011: possible nil pointer dereference (staticcheck)
		if *partition != expected {
		   ^
partition/gpt/partition_internal_test.go:58:6: SA5011(related information): this check suggests that the pointer can be nil (staticcheck)
		if partition == nil {
		   ^
diskfs.go:120:20: `firstblock` is unused (deadcode)
	defaultBlocksize, firstblock int = 512, 2048
	                  ^
filesystem/fat32/dos71bpb.go:32:2: `firstRemovableDrive` is unused (deadcode)
	firstRemovableDrive uint8 = 0x00
	^
filesystem/fat32/dos71bpb.go:34:2: `firstFixedDrive` is unused (deadcode)
	firstFixedDrive uint8 = 0x80
	^
filesystem/fat32/fat32.go:44:2: `minClusterSize` is unused (deadcode)
	minClusterSize       int        = 128
	^
filesystem/fat32/fat32.go:45:2: `maxClusterSize` is unused (deadcode)
	maxClusterSize       int        = 65529
	^
filesystem/fat32/fsinfosector.go:22:2: `unknownFreeDataClusterCount` is unused (deadcode)
	unknownFreeDataClusterCount uint32 = 0xffffffff
	^
filesystem/fat32/fsinfosector.go:24:2: `unknownlastAllocatedCluster` is unused (deadcode)
	unknownlastAllocatedCluster uint32 = 0xffffffff
	^
filesystem/fat32/fat32.go:411:13: Error return value of `file.ReadAt` is not checked (errcheck)
	file.ReadAt(b, int64(fatPrimaryStart)+int64(start))
	           ^
filesystem/fat32/fat32.go:417:13: Error return value of `file.ReadAt` is not checked (errcheck)
	file.ReadAt(b, int64(fatSecondaryStart)+int64(start))
	           ^
filesystem/fat32/fat32.go:624:17: Error return value of `fs.file.ReadAt` is not checked (errcheck)
		fs.file.ReadAt(tmpb, clusterStart)
		              ^
filesystem/fat32/fat32.go:902:17: Error return value of `fs.file.WriteAt` is not checked (errcheck)
	fs.file.WriteAt(b, fatPrimary+fs.start)
	               ^
filesystem/fat32/fat32.go:903:17: Error return value of `fs.file.WriteAt` is not checked (errcheck)
	fs.file.WriteAt(b, fatBackup+fs.start)
	               ^
filesystem/fat32/fat32.go:912:17: Error return value of `fs.file.WriteAt` is not checked (errcheck)
	fs.file.WriteAt(fsisBytes, int64(fsisPrimary)*int64(SectorSize512)+fs.start)
	               ^
filesystem/fat32/file.go:149: File is not `gofmt`-ed with `-s` (gofmt)
			_ , err := file.WriteAt(p[0:toWrite], int64(offset)+fs.start)
filesystem/fat32/directoryentry.go:180:58: `switchs` is a misspelling of `switches` (misspell)
			// need to break "byteLoop" else break will break the switchs
			                                                      ^
filesystem/fat32/directoryentry.go:87:2: `isDevice` is unused (structcheck)
	isDevice           bool
	^
filesystem/fat32/directoryentry.go:82:2: `isHidden` is unused (structcheck)
	isHidden           bool
	^
filesystem/fat32/directoryentry.go:93:2: `acccessRights` is unused (structcheck)
	acccessRights      accessRights
	^
filesystem/fat32/directoryentry.go:83:2: `isSystem` is unused (structcheck)
	isSystem           bool
	^
filesystem/fat32/directoryentry.go:81:2: `isReadOnly` is unused (structcheck)
	isReadOnly         bool
	^
filesystem/fat32/directoryentry.go:169:32: `parseDirEntries` - `f` is unused (unparam)
func parseDirEntries(b []byte, f *FileSystem) ([]*directoryEntry, error) {
                               ^
filesystem/fat32/directoryentry_internal_test.go:218:61: getValidDirectoryEntriesExtended - result 1 ([][]byte) is never used (unparam)
func getValidDirectoryEntriesExtended() ([]*directoryEntry, [][]byte, error) {
                                                            ^
filesystem/fat32/dos20bpb.go:44:41: (*dos20BPB).toBytes - result 1 (error) is always nil (unparam)
func (bpb *dos20BPB) toBytes() ([]byte, error) {
                                        ^
filesystem/fat32/fsinfosector.go:65:53: (*FSInformationSector).toBytes - result 1 (error) is always nil (unparam)
func (fsis *FSInformationSector) toBytes() ([]byte, error) {
                                                    ^
filesystem/fat32/table.go:39:40: tableFromBytes - result 1 (error) is always nil (unparam)
func tableFromBytes(b []byte) (*table, error) {
                                       ^
filesystem/fat32/table.go:62:34: (*table).bytes - result 1 (error) is always nil (unparam)
func (t *table) bytes() ([]byte, error) {
                                 ^
filesystem/fat32/directoryentry_internal_test.go:337:4: S1038: should use t.Logf(...) instead of t.Log(fmt.Sprintf(...)) (gosimple)
			t.Log(fmt.Sprintf("actual  : % x", output))
			^
filesystem/fat32/directoryentry_internal_test.go:338:4: S1038: should use t.Logf(...) instead of t.Log(fmt.Sprintf(...)) (gosimple)
			t.Log(fmt.Sprintf("expected: % x", tt.b))
			^
filesystem/fat32/fat32_internal_test.go:118:2: SA5001: should check returned error before deferring file.Close() (staticcheck)
	defer file.Close()
	^
partition/mbr/partiton.go:180:13: Error return value of `out.Write` is not checked (errcheck)
			out.Write(b[:read])
			         ^
partition/mbr/partiton.go:69:40: (*Partition).toBytes - result 1 (error) is always nil (unparam)
func (p *Partition) toBytes() ([]byte, error) {
                                       ^
partition/mbr/partiton.go:89:35: `partitionFromBytes` - `logicalSectorSize` always receives `logicalSectorSize` (`512`) (unparam)
func partitionFromBytes(b []byte, logicalSectorSize, physicalSectorSize int) (*Partition, error) {
                                  ^
partition/mbr/table.go:58:27: `(*Table).initTable` - `size` is unused (unparam)
func (t *Table) initTable(size int64) {
                          ^
partition/mbr/table.go:83:31: `tableFromBytes` - `logicalBlockSize` is unused (unparam)
func tableFromBytes(b []byte, logicalBlockSize, physicalBlockSize int) (*Table, error) {
                              ^
partition/mbr/partiton.go:70:20: S1019: should use make([]byte, partitionEntrySize) instead (gosimple)
	b := make([]byte, partitionEntrySize, partitionEntrySize)
	                  ^
partition/mbr/partiton.go:126:20: S1019: should use make([]byte, pss) instead (gosimple)
	b := make([]byte, pss, pss)
	                  ^
partition/mbr/partiton.go:168:20: S1019: should use make([]byte, pss) instead (gosimple)
	b := make([]byte, pss, pss)
	                  ^
partition/mbr/table.go:125:20: S1019: should use make([]byte, mbrSize) instead (gosimple)
	b := make([]byte, mbrSize, mbrSize)
	                  ^
partition/mbr/table.go:179:34: S1019: should use make([]part.Partition, len(t.Partitions)) instead (gosimple)
	parts := make([]part.Partition, len(t.Partitions), len(t.Partitions))
	                                ^
partition/mbr/partition_internal_test.go:24:21: S1019: should use make([]byte, partitionEntrySize - 1) instead (gosimple)
		b := make([]byte, partitionEntrySize-1, partitionEntrySize-1)
		                  ^
partition/mbr/partition_internal_test.go:39:21: S1019: should use make([]byte, partitionEntrySize + 1) instead (gosimple)
		b := make([]byte, partitionEntrySize+1, partitionEntrySize+1)
		                  ^
partition/mbr/partition_internal_test.go:54:21: S1019: should use make([]byte, partitionEntrySize) instead (gosimple)
		b := make([]byte, partitionEntrySize, partitionEntrySize)
		                  ^
partition/mbr/partition_internal_test.go:284:21: S1019: should use make([]byte, 2 * 512) instead (gosimple)
		b := make([]byte, 2*512, 2*512)
		                  ^
partition/mbr/table_internal_test.go:44:21: S1019: should use make([]byte, 512 - 1) instead (gosimple)
		b := make([]byte, 512-1, 512-1)
		                  ^
partition/mbr/partiton.go:44:3: S1004: should use bytes.Equal(b1[8:12], b2[8:12]) instead (gosimple)
		bytes.Compare(b1[8:12], b2[8:12]) == 0 &&
		^
partition/mbr/partiton.go:45:3: S1004: should use bytes.Equal(b1[12:16], b2[12:16]) instead (gosimple)
		bytes.Compare(b1[12:16], b2[12:16]) == 0
		^
partition/mbr/table.go:91:5: S1004: should use !bytes.Equal(mbrSignature, getMbrSignature()) instead (gosimple)
	if bytes.Compare(mbrSignature, getMbrSignature()) != 0 {
	   ^
partition/mbr/partition_internal_test.go:198:6: S1004: should use !bytes.Equal(b.Bytes(), b2) instead (gosimple)
		if bytes.Compare(b.Bytes(), b2) != 0 {
		   ^
partition/mbr/partition_internal_test.go:342:6: S1004: should use !bytes.Equal(b2, b) instead (gosimple)
		if bytes.Compare(b2, b) != 0 {
		   ^
partition/mbr/partition_internal_test.go:94:26: SA4022: the address of a variable cannot be nil (staticcheck)
		if partition == nil && &expected != nil || !partition.Equal(&expected) {
		                       ^
disk/disk_unix.go:1: File is not `gofmt`-ed with `-s` (gofmt)
// +build aix darwin dragonfly freebsd linux netbsd openbsd solaris
disk/disk.go:183:2: missing cases in switch of type filesystem.Type: TypeSquashfs (exhaustive)
	switch spec.FSType {
	^
filesystem/fat32/fat32_test.go:495:21: Error return value of `readWriter.Seek` is not checked (errcheck)
					readWriter.Seek(0, 0)
					               ^
filesystem/fat32/fat32_test.go:554:21: Error return value of `readWriter.Seek` is not checked (errcheck)
					readWriter.Seek(0, 0)
					               ^
filesystem/fat32/fat32_test.go:556:21: Error return value of `readWriter.Seek` is not checked (errcheck)
					readWriter.Seek(0, 0)
					               ^
filesystem/fat32/fat32_test.go:347:1: cyclomatic complexity 90 of func `TestFat32OpenFile` is high (> 30) (gocyclo)
func TestFat32OpenFile(t *testing.T) {
^
filesystem/fat32/fat32_test.go:332:9: S1009: should omit nil check; len() for []io/fs.FileInfo is defined as zero (gosimple)
			case output != nil && len(output) > 0 && output[0].IsDir() != tt.isDir:
			     ^
filesystem/fat32/fat32_test.go:334:9: S1009: should omit nil check; len() for []io/fs.FileInfo is defined as zero (gosimple)
			case output != nil && len(output) > 0 && output[0].Name() != tt.name:
			     ^
filesystem/fat32/fat32_test.go:114:13: test helper function should start from t.Helper() (thelper)
	runTest := func(t *testing.T, post, pre int64, fatFunc func(util.File, int64, int64, int64) (*fat32.FileSystem, error)) {
	           ^
filesystem/fat32/fat32_test.go:194:13: test helper function should start from t.Helper() (thelper)
	runTest := func(t *testing.T, pre, post int64) {
	           ^
filesystem/fat32/fat32_test.go:242:13: test helper function should start from t.Helper() (thelper)
	runTest := func(t *testing.T, pre, post int64) {
	           ^
filesystem/iso9660/iso9660_test.go:307:14: Error return value of `os.MkdirAll` is not checked (errcheck)
		os.MkdirAll(existPathWs, 0755)
		           ^
filesystem/iso9660/iso9660_test.go:312:20: Error return value of `ioutil.WriteFile` is not checked (errcheck)
			ioutil.WriteFile(filename, []byte(contents), 0644)
			                ^
filesystem/iso9660/iso9660_test.go:410:15: Error return value of `os.MkdirAll` is not checked (errcheck)
			os.MkdirAll(path.Join(ws, subdir), 0755)
			           ^
filesystem/iso9660/iso9660_test.go:414:21: Error return value of `ioutil.WriteFile` is not checked (errcheck)
				ioutil.WriteFile(path.Join(ws, subdir, filename), []byte(content), 0644)
				                ^
filesystem/iso9660/iso9660_test.go:490:23: Error return value of `ioutil.WriteFile` is not checked (errcheck)
						ioutil.WriteFile(filepath, []byte(baseContent), 0644)
						                ^
filesystem/iso9660/finalize_test.go:102:1: cyclomatic complexity 49 of func `TestFinalize9660` is high (> 30) (gocyclo)
func TestFinalize9660(t *testing.T) {
^
filesystem/iso9660/iso9660_test.go:333:1: cyclomatic complexity 45 of func `TestIso9660OpenFile` is high (> 30) (gocyclo)
func TestIso9660OpenFile(t *testing.T) {
^
filesystem/iso9660/finalize_test.go:340:1: cyclomatic complexity 31 of func `TestFinalizeRockRidge` is high (> 30) (gocyclo)
func TestFinalizeRockRidge(t *testing.T) {
^
filesystem/iso9660/iso9660_test.go:322:22: G601: Implicit memory aliasing in for loop. (gosec)
			fis = append(fis, &e)
			                  ^
filesystem/iso9660/iso9660_test.go:508:41: SA1019: os.SEEK_SET has been deprecated since Go 1.7: Use io.SeekStart, io.SeekCurrent, and io.SeekEnd. (staticcheck)
							offset, err = readWriter.Seek(0, os.SEEK_SET)
							                                 ^
filesystem/squashfs/finalize.go:1293:6: `dirMapToList` is unused (deadcode)
func dirMapToList(dirMap map[string]*finalizeFileInfo) []*finalizeFileInfo {
     ^
filesystem/squashfs/fragment.go:9:2: `fragmentEntriesPerBlock` is unused (deadcode)
	fragmentEntriesPerBlock = 512
	^
filesystem/squashfs/fragment.go:10:2: `fragmentEntrySize` is unused (deadcode)
	fragmentEntrySize       = 16
	^
filesystem/squashfs/fragment.go:19:6: `fragmentTable` is unused (deadcode)
type fragmentTable struct {
     ^
filesystem/squashfs/inode.go:866:6: `idTable` is unused (deadcode)
type idTable []uint32
     ^
filesystem/squashfs/const_internal_test.go:53:2: `testLargeDirEntryCount` is unused (deadcode)
	testLargeDirEntryCount                 = 252
	^
filesystem/squashfs/const_internal_test.go:88:2: `testDirectoryDataSize` is unused (deadcode)
	testDirectoryDataSize = testFragmentOffset - testDirectoryOffset
	^
filesystem/squashfs/const_internal_test.go:89:2: `testIDTableStart` is unused (deadcode)
	testIDTableStart      = 0x5092c8 + 2 - testMetaOffset
	^
filesystem/squashfs/const_internal_test.go:90:2: `testIDTableEnd` is unused (deadcode)
	testIDTableEnd        = 0x5092d6 - testMetaOffset
	^
filesystem/squashfs/const_internal_test.go:92:2: `testXattrIDEnd` is unused (deadcode)
	testXattrIDEnd        = testValidSuperblockUncompressed.xattrTableStart - testMetaOffset
	^
filesystem/squashfs/const_internal_test.go:93:2: `testXattrMetaStart` is unused (deadcode)
	testXattrMetaStart    = 0x5092de + 2 - testMetaOffset
	^
filesystem/squashfs/const_internal_test.go:94:2: `testXattrMetaEnd` is unused (deadcode)
	testXattrMetaEnd      = testXattrIDStart
	^
filesystem/squashfs/const_internal_test.go:98:2: `testInodeDataLength` is unused (deadcode)
	testInodeDataLength = 16766
	^
filesystem/squashfs/const_internal_test.go:202:6: `testGetFilesystemRoot` is unused (deadcode)
func testGetFilesystemRoot() []*directoryEntry {
     ^
filesystem/squashfs/directory_internal_test.go:40: 40-52 lines are duplicate of `filesystem/squashfs/fragment_internal_test.go:22-34` (dupl)
	for i, tt := range tests {
		header, err := parseDirectoryHeader(tt.b)
		switch {
		case (err == nil && tt.err != nil) || (err != nil && tt.err == nil) || (err != nil && tt.err != nil && !strings.HasPrefix(err.Error(), tt.err.Error())):
			t.Errorf("%d: mismatched error, actual then expected", i)
			t.Logf("%v", err)
			t.Logf("%v", tt.err)
		case (header == nil && tt.header != nil) || (header != nil && tt.header == nil) || (header != nil && tt.header != nil && *header != *tt.header):
			t.Errorf("%d: mismatched header, actual then expected", i)
			t.Logf("%v", header)
			t.Logf("%v", tt.header)
		}
	}
filesystem/squashfs/fragment_internal_test.go:22: 22-34 lines are duplicate of `filesystem/squashfs/superblock_internal_test.go:57-69` (dupl)
	for i, tt := range testFragmentEntries {
		entry, err := parseFragmentEntry(tt.b)
		switch {
		case (err == nil && tt.err != nil) || (err != nil && tt.err == nil) || (err != nil && tt.err != nil && !strings.HasPrefix(err.Error(), tt.err.Error())):
			t.Errorf("%d: mismatched error, actual then expected", i)
			t.Logf("%v", err)
			t.Logf("%v", tt.err)
		case (entry == nil && tt.entry != nil) || (entry != nil && tt.entry == nil) || (entry != nil && tt.entry != nil && *entry != *tt.entry):
			t.Errorf("%d: mismatched header, actual then expected", i)
			t.Logf("%v", entry)
			t.Logf("%v", tt.entry)
		}
	}
filesystem/squashfs/superblock_internal_test.go:57: 57-69 lines are duplicate of `filesystem/squashfs/directory_internal_test.go:40-52` (dupl)
	for i, tt := range tests {
		flags, err := parseFlags(tt.b)
		switch {
		case (err == nil && tt.err != nil) || (err != nil && tt.err == nil) || (err != nil && tt.err != nil && !strings.HasPrefix(err.Error(), tt.err.Error())):
			t.Errorf("%d: mismatched error, actual then expected", i)
			t.Logf("%v", err)
			t.Logf("%v", tt.err)
		case (flags == nil && tt.flags != nil) || (flags != nil && tt.flags == nil) || (flags != nil && tt.flags != nil && *flags != *tt.flags):
			t.Errorf("%d: mismatched results, actual then expected", i)
			t.Logf("%v", flags)
			t.Logf("%v", tt.flags)
		}
	}
filesystem/squashfs/inode_internal_test.go:237: 237-251 lines are duplicate of `filesystem/squashfs/inode_internal_test.go:311-325` (dupl)
	t.Run("parse", func(t *testing.T) {
		for i, tt := range tests {
			fl, _, err := parseBasicFile(tt.b, int(testValidBlocksize))
			switch {
			case (err == nil && tt.err != nil) || (err != nil && tt.err == nil) || (err != nil && tt.err != nil && !strings.HasPrefix(err.Error(), tt.err.Error())):
				t.Errorf("%d: mismatched error, actual then expected", i)
				t.Logf("%v", err)
				t.Logf("%v", tt.err)
			case (fl == nil && tt.file != nil) || (fl != nil && tt.file == nil) || (fl != nil && tt.file != nil && !fl.equal(*tt.file)):
				t.Errorf("%d: mismatched results, actual then expected", i)
				t.Logf("%#v", *fl)
				t.Logf("%#v", *tt.file)
			}
		}
	})
filesystem/squashfs/inode_internal_test.go:311: 311-325 lines are duplicate of `filesystem/squashfs/inode_internal_test.go:237-251` (dupl)
	t.Run("parse", func(t *testing.T) {
		for i, tt := range tests {
			fl, _, err := parseExtendedFile(tt.b, int(testValidBlocksize))
			switch {
			case (err == nil && tt.err != nil) || (err != nil && tt.err == nil) || (err != nil && tt.err != nil && !strings.HasPrefix(err.Error(), tt.err.Error())):
				t.Errorf("%d: mismatched error, actual then expected", i)
				t.Logf("%v", err)
				t.Logf("%v", tt.err)
			case (fl == nil && tt.file != nil) || (fl != nil && tt.file == nil) || (fl != nil && tt.file != nil && !fl.equal(*tt.file)):
				t.Errorf("%d: mismatched results, actual then expected", i)
				t.Logf("%#v", *fl)
				t.Logf("%#v", *tt.file)
			}
		}
	})
filesystem/squashfs/finalize.go:425:10: Error return value of `os.Chdir` is not checked (errcheck)
	os.Chdir(workspace)
	        ^
filesystem/squashfs/finalize.go:429:15: Error return value of `filepath.Walk` is not checked (errcheck)
	filepath.Walk(".", func(fp string, fi os.FileInfo, err error) error {
	             ^
filesystem/squashfs/finalize.go:1481:11: Error return value is not checked (errcheck)
			dir := inBody.(*basicDirectory)
			       ^
filesystem/squashfs/finalize.go:1486:11: Error return value is not checked (errcheck)
			dir := inBody.(*extendedDirectory)
			       ^
filesystem/squashfs/metadatablock.go:77:10: Error return value of `r.ReadAt` is not checked (errcheck)
	r.ReadAt(b, location)
	        ^
filesystem/squashfs/inode_internal_test.go:392:28: `withh` is a misspelling of `with` (misspell)
	// when we have more data withh which to work
	                          ^
filesystem/squashfs/inode_internal_test.go:400:28: `withh` is a misspelling of `with` (misspell)
	// when we have more data withh which to work
	                          ^
filesystem/squashfs/inode_internal_test.go:409:28: `withh` is a misspelling of `with` (misspell)
	// when we have more data withh which to work
	                          ^
filesystem/squashfs/finalize.go:1345:2: Consider preallocating `pairs` (prealloc)
	var pairs []string
	^
filesystem/squashfs/finalize.go:1371:2: Consider preallocating `entries` (prealloc)
	var (
	^
filesystem/squashfs/fragment.go:20:2: `entries` is unused (structcheck)
	entries []*fragmentEntry
	^
filesystem/squashfs/inode_internal_test.go:15:2: `entries` is unused (structcheck)
	entries  []*directoryEntry
	^
filesystem/squashfs/inode_internal_test.go:16:2: `err` is unused (structcheck)
	err      error
	^
filesystem/squashfs/finalizefileinfo.go:28:2: `bytes` is unused (structcheck)
	bytes             [][]byte
	^
filesystem/squashfs/finalizefileinfo.go:20:2: `recordSize` is unused (structcheck)
	recordSize        uint8
	^
filesystem/squashfs/finalizefileinfo.go:31:2: `content` is unused (structcheck)
	content           []byte
	^
filesystem/squashfs/finalizefileinfo.go:19:2: `location` is unused (structcheck)
	location          uint32
	^
filesystem/squashfs/finalizefileinfo.go:18:2: `target` is unused (structcheck)
	target            string
	^
filesystem/squashfs/squashfs.go:695:49: `parseXattrsTable` - `offset` is unused (unparam)
func parseXattrsTable(bUIDXattr, bIndex []byte, offset uint64, c Compressor) (*xAttrTable, error) {
                                                ^
filesystem/squashfs/util.go:20:42: universalizePath - result 1 (error) is always nil (unparam)
func universalizePath(p string) (string, error) {
                                         ^
filesystem/squashfs/squashfs.go:307:3: missing cases in switch of type inodeType: inodeBasicDirectory (exhaustive)
		switch iType {
		^
filesystem/squashfs/inode.go:661:2: S1021: should merge variable declaration with assignment on next line (gosimple)
	var devNum uint32
	^
filesystem/squashfs/compressor_internal_test.go:62:8: S1004: should use !bytes.Equal(out, testCompressUncompressed) instead (gosimple)
		case bytes.Compare(out, testCompressUncompressed) != 0:
		     ^
filesystem/squashfs/squashfs.go:463:10: ineffectual assignment to err (ineffassign)
	header, err := parseInodeHeader(uncompressed)
	        ^
filesystem/squashfs/finalize.go:450:3: SA4009(related information): assignment to err (staticcheck)
		xattrNames, err := xattr.List(fp)
		^
filesystem/squashfs/compressor.go:69:2: SA9004: only the first constant in this group has an explicit type (staticcheck)
	GzipDefault          GzipStrategy = 0x1
	^
filesystem/squashfs/compressor.go:152:2: SA9004: only the first constant in this group has an explicit type (staticcheck)
	XzFilterX86      XzFilter = 0x1
	^
filesystem/squashfs/inode.go:13:2: SA9004: only the first constant in this group has an explicit type (staticcheck)
	inodeBasicDirectory    inodeType = 1
	^
filesystem/squashfs/compressor.go:118:2: SA4005: ineffective assignment to field CompressorGzip.CompressionLevel (staticcheck)
	c.CompressionLevel = binary.LittleEndian.Uint32(b[0:4])
	^
filesystem/squashfs/compressor.go:119:2: SA4005: ineffective assignment to field CompressorGzip.WindowSize (staticcheck)
	c.WindowSize = binary.LittleEndian.Uint16(b[4:6])
	^
filesystem/squashfs/compressor.go:127:2: SA4005: ineffective assignment to field CompressorGzip.Strategies (staticcheck)
	c.Strategies = strategies
	^
filesystem/squashfs/compressor.go:200:2: SA4005: ineffective assignment to field CompressorXz.DictionarySize (staticcheck)
	c.DictionarySize = binary.LittleEndian.Uint32(b[0:4])
	^
filesystem/squashfs/compressor.go:208:2: SA4005: ineffective assignment to field CompressorXz.ExecutableFilters (staticcheck)
	c.ExecutableFilters = filters
	^
filesystem/squashfs/compressor.go:272:2: SA4005: ineffective assignment to field CompressorLz4.version (staticcheck)
	c.version = version
	^
filesystem/squashfs/compressor.go:280:2: SA4005: ineffective assignment to field CompressorLz4.flags (staticcheck)
	c.flags = flagMap
	^
filesystem/squashfs/compressor.go:318:2: SA4005: ineffective assignment to field CompressorZstd.level (staticcheck)
	c.level = level
	^
filesystem/squashfs/squashfs_test.go:169:14: Error return value of `os.MkdirAll` is not checked (errcheck)
		os.MkdirAll(existPathWs, 0755)
		           ^
filesystem/squashfs/squashfs_test.go:184:22: G601: Implicit memory aliasing in for loop. (gosec)
			fis = append(fis, &e)
			                  ^
partition/partition_test.go:36:14: Error return value of `f.Truncate` is not checked (errcheck)
			f.Truncate(10 * 1024 * 1024)
			          ^
partition/mbr/table_test.go:138:3: SA5001: should check returned error before deferring f.Close() (staticcheck)
		defer f.Close()
		^
partition/mbr/table_test.go:193:3: SA5001: should check returned error before deferring mbr.Close() (staticcheck)
		defer mbr.Close()
		^
```

</details>

deitch commented:
I see 271 remaining lint errors, on top of the dozens you fixed in this PR. I want to merge this in, but not at the cost of breaking CI. Which means fixing them. I appreciate your adding these; do you want to add more fixes?

dreddsa5dies commented:
Fixing all alerts is a big deal. Which will take a lot of time, during which even more alerts will accumulate (with the "old" verification method).
The following order is proposed: I will comment out the extra linters in the configuration file so that I can pass CI now.
You have received the necessary information about alerts and can discuss further actions with colleagues.

deitch commented:
Replaced by #149
